benchmark/vochain_benchmark_test.go

```diff
@@ -27,214 +27,133 @@ const (
+	// setup the basic parties
+	signerPubHex, _ := dvoteServer.Signer.HexString()
+	signerPub := testutil.Hex2byte(b, signerPubHex)
 	entityID := signerPub[:types.EntityIDsize]
 	processID := testutil.Hex2byte(b, hexProcessID)
-	processData := &models.Process{
-		EntityId:     entityID,
-		CensusRoot:   censusRoot,
-		CensusOrigin: models.CensusOrigin_OFF_CHAIN_TREE,
-		BlockCount:   numberOfBlocks,
-		ProcessId:    processID,
-		StartBlock:   *resp.Height + 1,
-		Status:       models.ProcessStatus_READY,
-		EnvelopeType: &models.EnvelopeType{},
-		Mode:         &models.ProcessMode{AutoStart: true},
-		VoteOptions:  &models.ProcessVoteOptions{MaxCount: 1, MaxValue: 1},
-	}
-	process := &models.NewProcessTx{
-		Txtype:  models.TxType_NEW_PROCESS,
-		Nonce:   0,
-		Process: processData,
-	}
 
-	stx := &models.SignedTx{}
-	stx.Tx, err = proto.Marshal(&models.Tx{Payload: &models.Tx_NewProcess{NewProcess: process}})
-	if err != nil {
-		b.Fatal("cannot marshal process")
-	}
-	stx.Signature, err = dvoteServer.Signer.SignVocdoniTx(stx.Tx, dvoteServer.VochainAPP.ChainID())
-	if err != nil {
-		b.Fatalf("cannot sign oracle tx: %s", err)
-	}
-
-	reset(req)
-	req.Payload, err = proto.Marshal(stx)
-	if err != nil {
-		b.Fatalf("error marshaling process tx: %v", err)
-	}
-	resp = doRequest("submitRawTx", nil)
-	if !resp.Ok {
-		b.Fatalf("error broadcasting process tx: %s", resp.Message)
-	} else {
-		log.Info("process transaction sent")
-	}
-
-	// check if process is created
-	log.Infof("check if process was created")
-	reset(req)
-	req.ProcessID = processID
-	failures := 20
-	for {
-		resp = doRequest("getProcessInfo", nil)
-		if resp.Ok {
-			break
-		}
-		failures--
-		if failures == 0 {
-			b.Fatalf("processID does not exist in the blockchain")
-		}
-		time.Sleep(time.Second)
-	}
+	// create census and process
+	censusRoot, _, processID := createDistributedCensusAndProcess(b, cl, dvoteServer, keySet, entityID)
+	waitProcess(b, cl, processID)
 
 	// send votes in parallel
+	log.Infof("sending %d votes in parallel", b.N)
 	count := int32(0)
 	b.ResetTimer()
 
 	b.RunParallel(func(pb *testing.PB) {
-		// Create websocket client
 		cl, err := client.New(host)
 		if err != nil {
 			b.Fatal(err)
 		}
 		for pb.Next() {
-			voteBench(b,
+			signer := keySet[count]
+			siblings, value, err := cl.GetProof(signer.PublicKey(), censusRoot, false)
+			if err != nil {
+				b.Fatalf("could not get proof for %s: %s", signer.AddressString(), err)
+			}
+			proof := &client.Proof{
+				Siblings: siblings,
+				Value:    value,
+			}
+
+			sendVote(b,
 				cl,
-				keySet[atomic.AddInt32(&count, 1)],
-				censusRoot,
-				processID)
+				dvoteServer.VochainAPP.ChainID(),
+				signer,
+				proof,
+				processID,
+			)
+			atomic.AddInt32(&count, 1)
 		}
 	})
 }
 
-func voteBench(b *testing.B, cl *client.Client, s *ethereum.SignKeys,
-	censusRoot, processID []byte) {
-	// API requests
+func sendVote(b *testing.B, cl *client.Client, chainID string, s *ethereum.SignKeys,
```

Comment:
I assume you're not returning an error here and in the other functions because old code do the things that way, but  what do you think if we wrap the error and we call the fatals on the main function of the bench? I think is cleaner

Reply:
Yes, I was following the old code's way.

The code leading up to the vote submission fails with a Fatal, but vote submission fails with an Error. If we move this upwards, this distinction will be lost. Is it important? Perhaps everything should be fatal?

Reply:
Changed it to this:
```
err = sendVote(b,
    cl,
    dvoteServer.VochainAPP.ChainID(),
    signer,
    proof,
    processID,
)
if err != nil && strings.HasPrefix(err.Error(), "error submitting vote") {
    b.Error(err)
} else if err != nil {
    b.Fatal(err)
}
```

Reply:
Basically, my strategy is: sacrifice code 'tidiness' and DRY-ness in favour of clarity of purpose when reading higher level code.